core/storage/filestore.go

```diff
@@ -0,0 +1,30 @@
+package storage
+
+// Filestore is a interface for file storage
+type Filestore interface {
```

Comment:
Let's put `ctx context.Context` as a parameter to all the methods. We'll need that for all network Filestores.

Reply:
I'm not familiar with this specific pattern in go. I've added it as I think you meant it. Let me know.

Reply:
This is right.